RULES/cc/cc.go

```diff
@@ -20,7 +21,32 @@ type objectFile struct {
 func (obj objectFile) Build(ctx core.Context) {
 	toolchain := toolchainOrDefault(obj.Toolchain)
 	depfile := obj.Out.WithExt("d")
-	cmd := toolchain.ObjectFile(obj.Out, depfile, obj.Flags, obj.Includes, obj.Src)
+
+	cmd := ""
+	flags := []string{}
+
+	switch filepath.Ext(obj.Src.Absolute()) {
+	case ".cc":
+		cmd = toolchain.CxxCompiler()
+		flags = append(toolchain.CxxFlags(), "-pipe", "-c", "-MD", "-MF", fmt.Sprintf("%q", depfile))
+	case ".c":
+		cmd = toolchain.CCompiler()
+		flags = append(toolchain.CFlags(), "-pipe", "-c", "-MD", "-MF", fmt.Sprintf("%q", depfile))
+	case ".S":
+		cmd = toolchain.Assembler()
+		flags = append(toolchain.AsFlags(), "-c")
+	default:
+		panic("Unknown source extension for cc toolchain '" + filepath.Ext(obj.Src.Absolute()) + "'")
```

Comment:
Can you use `core.Fatal` instead? That e.g. suppresses output of errors when generating completions and will output the target that caused the error.